src/monitors/turbFlowStats.cpp

```diff
@@ -112,12 +112,13 @@ PetscErrorCode ablate::monitors::TurbFlowStats::MonitorTurbFlowStats(TS ts, Pets
                     monitorPt[monitorFields[FieldPlacements::densitySum].offset] += densLoc;
                     monitorPt[monitorFields[FieldPlacements::densityDtSum].offset] += densLoc * dt;
 
-                    // Set the offset
-                    PetscInt offset = monitorFields[FieldPlacements::fieldsStart + f].offset;
+                    const auto& field = monitor->GetSolver()->GetSubDomain().GetField(monitor->fieldNames[f]);
 
                     // March over each field component
-                    for (int p = 0; p < monitorFields[f].numberComponents; p++) {
-                        offset += SectionLabels::END * p;
```

Comment:
Was this only broken when dim > 2? This change makes sense to me but it took a while to figure out the math.  Can you add a comment that explains the offset and how the components are arranged?

Reply:
Both 3D and 2D actually. The integration test, which is 2D gives garbage for the second velocity.

Reply:
Tried to explain it. let me know if its is confusing